### PR TITLE
bugfix: failed to get `host` in health check configuration.

### LINKED ISSUE #1869: bug: failed to get `host` in health check configuration.
## code location
https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L64
https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L95
https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L198
https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L201
https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L206
##
refer to the health check configuration:
https://github.com/apache/incubator-apisix/blob/master/doc/health-check.md
`host` should be from `checks.active.host` instead of `checks.host`

### PR DESCRIPTION
**Pull request by shuaijinchao**
### Pre-submission checklist:
FIX #1869 

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by shuaijinchao** [comment]
@membphis the test case cannot get the warn log thrown by the health check, please help to check.

**Comment by membphis** [comment]
I check the source code, and I found two points.

> the healthcheck was ignored

1. if the count of upstream node is 1, it will return the upstream node directly, the healthcheck was ignored. 

https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L180-L185

I think we can create a new PR to fix this bug first.

> The health check prints the log only when the status of the upstream node is changed

The default status is SUCCESS, if the status is SUCCESS always, we never can get a log like `healthy SUCCESS`.

**Comment by shuaijinchao** [comment]
> I check the source code, and I found two points.
> 
> > the healthcheck was ignored
> 
> 1. if the count of upstream node is 1, it will return the upstream node directly, the healthcheck was ignored.
> 
> https://github.com/apache/incubator-apisix/blob/master/apisix/balancer.lua#L180-L185
> 
> I think we can create a new PR to fix this bug first.
> 
> > The health check prints the log only when the status of the upstream node is changed
> 
> The default status is SUCCESS, if the status is SUCCESS always, we never can get a log like `healthy SUCCESS`.

1. If the number of nodes is 1 and the node is a faulty node, the health check is of little significance at this time, and APISIX will also have a retry mechanism.
2. adding a faulty node to trigger the status change of the health check, the verification of the test case is completed.

Thanks @membphis for your help.

File: t/node/healthcheck.t [hunk 1]
```diff
@@ -548,3 +548,73 @@ code: 200 body: passed
...
+--- config
+    location /t {
+        content_by_lua_block {
+            local http = require "resty.http"
+            local uri = "http://127.0.0.1:" .. ngx.var.server_port
+                        .. "/server_port"
+
+            local httpc = http.new()
+            local res, err = httpc:request_uri(uri, {method = "GET", keepalive = false})
+            ngx.say(res.status)
```
Comment (membphis) [inline t/node/healthcheck.t:610]: add a sleep time is safer, eg: `ngx.sleep(2)`

Checking the status of upstream nodes is asynchronous.

Comment (shuaijinchao) [reply t/node/healthcheck.t:613]: already fixed

**Comment by membphis** [comment]
@shuaijinchao 

> Is this PR backward compatible?

I think this PR is not backward compatible. please  confirm this.

**Comment by shuaijinchao** [comment]
> @shuaijinchao
> 
> > Is this PR backward compatible?
> 
> I think this PR is not backward compatible. please confirm this.

@membphis  this is a bug fix, using the previous data structure, so it is backward compatible.

**Comment by membphis** [comment]
> @membphis this is a bug fix, using the previous data structure, so it is backward compatible.

ok, got it. let us merge this PR.

**Comment by membphis** [comment]
@shuaijinchao merged, many thx

**Pull request merged**